projects/rocprofiler-systems/source/lib/common/environment.hpp

```diff
@@ -177,5 +179,48 @@ struct ROCPROFSYS_INTERNAL_API env_config
         return setenv(env_name.c_str(), env_value.c_str(), override);
     }
 };
+
+inline std::string
+discover_llvm_libdir_for_ompt(bool verbose = false)
+{
+    auto strip = [](std::string s) {
+        if(!s.empty() && s.back() == '/') s.pop_back();
+        return s;
+    };
+
+    // Common ROCm envs
+    const auto rocm_dir  = strip(get_env<std::string>("ROCM_PATH", "/opt/rocm"));
+    const auto rocmv_dir = strip(get_env<std::string>("ROCmVersion_DIR", ""));
+
+    std::vector<std::string> candidates;
+    candidates.reserve(6);
```

Comment:
```suggestion
    std::vector<std::string> candidates(6);
```
nitpick: Will this simplify things a little?

Reply:
I kept reserve(6) intentionally so the vector starts empty but has capacity pre-allocated - this avoids six default-constructed empty strings that would occur with vector(6) and lets us emplace_back() cleanly.